drivers/mbed_lib.json

```diff
@@ -12,6 +12,24 @@
         "spi_count_max": {
             "help": "The maximum number of SPI peripherals used at the same time. Determines RAM allocated for SPI peripheral management. If null, limit determined by hardware.",
             "value": null
+        },
+        "qspi_io0": {
```

Comment:
We have `QSPI_FLASH1_IO0` and now introducing `QSPI_FLASH1_IO0` ?

Why are not these in PinNames and defined for all boards having QSPI? There are QSPI pins or not? They should provide default pins,  that could be overriden in tests/apps via config. Is this what is it adding?

Reply:
Goal is to use an external QSPI,
and add pins value in mbed_app.json file

Reply:
I understand now. This is a new approach here in this drivers config. This brings QSPI definitions even for a board that does not have it. Shouldn' t this be by default null, not even created if not used , and a target would override it and set?

@maciejbocianski what do you think?

Reply:
Goal of this approach is to be full compatible with current implementation.
There is no impact with current targets with and without QSPI pins

Reply:
The only impact having QSPI defined always even if QSPI is not available?

Reply:
4 new define added and not used in the mbed_config.h file